File: src/main/java/types/Channel.java
package types;

import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "channels")
public class Channel {

    @Id
    private ObjectId channelId;
    private String channelName;

    public String getChannelName() { return this.channelName; }

    public void setChannelName(String channelName) {
        this.channelName = channelName;
    }
}

File: src/main/java/io/github/thewilly/conductor/server/types/Channel.kt
package io.github.thewilly.conductor.server.types

import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "channels")
data class Channel(val channelName: String) {

    @Id
    private val channelId: ObjectId? = null
}
